File: packages/nativescript-image-cache-it/src-native/imagecacheit/src/main/java/com/github/triniwiz/imagecacheit/ImageViewProgressListener.java
package com.github.triniwiz.imagecacheit;

/**
 * Created by triniwiz on 3/5/20
 */
public interface ImageViewProgressListener {

    void onProgress(String key, long bytesRead, long expectedLength);

    /**
     * Control how often the listener needs an update. 0% and 100% will always be dispatched.
     *
     * @return in percentage (0.2 = call {@link #onProgress} around every 0.2 percent of progress)
     */
    float getGranularityPercentage();
}

File: packages/nativescript-image-cache-it/src-native/imagecacheit/src/main/java/com/github/triniwiz/imagecacheit/ImageViewProgressListener.kt
package com.github.triniwiz.imagecacheit

/**
 * Created by triniwiz on 3/5/20
 */
interface ImageViewProgressListener {
  fun onProgress(key: String?, bytesRead: Long, expectedLength: Long)

  /**
   * Control how often the listener needs an update. 0% and 100% will always be dispatched.
   *
   * @return in percentage (0.2 = call [.onProgress] around every 0.2 percent of progress)
   */
  val granularityPercentage: Float
}
